src/pl/michal/Bank.java

```diff
@@ -22,7 +25,32 @@ public void transfer(BankAccount from, BankAccount to, int amount) {
         account += FEE;
     }
 
+    public void transfer(BankAccount from, BankAccount to, double amount) {
+        if (amount <= 0) {
+            this.transfer(from, to, 0);
+        } else {
+            int fromNO = from.getAccountNO(),
+                    toNO = to.getAccountNO();
+
+            System.out.println(String.format("Transfer %.3f zl from %s account to %s account", amount, fromNO, toNO));
+
+            try {
+                from.withdraw(amount + FEE);
+
+                mySQLaccess.InsertBankTransaction(fromNO, toNO, amount, FEE);
+                mySQLaccess.FinishTransaction();
```

Comment:
Wywołanie InsertBankTransaction oraz FinishTransaction powinno być w jednej metodzie DAO (MySQLaccess). Rozdzielenie tego i wywołanie osobne w tym miejscu nie jest dobre, ale to pogadamy przy okazji o zrobieniu lepszej struktury organizacyjnej/archutektury dla bazy danych.